azsqs/queue.go

```diff
@@ -112,7 +112,7 @@ func (q *Queue) Options() *taskq.QueueOptions {
 	return q.opt
 }
 
-func (q *Queue) Consumer() *taskq.Consumer {
+func (q *Queue) Consumer() taskq.QueueConsumer {
```

Comment:
This looks unneeded, right?

Reply:
True, but `Consumer` signature was changed in this PR

```diff
-	Consumer() *Consumer
+	Consumer() QueueConsumer
```

So it should work if we revert that change too. The question is do you need that change for mocking?

Reply:
Yes otherwise it's not possible to mock. For example when your code relies on the `Queue` interface you can replace it with a mock, but as soon your code is using `queue.Consumer()` you get in trouble as you can't return a mock here anymore.

Reply:
Makes sense - thank you.